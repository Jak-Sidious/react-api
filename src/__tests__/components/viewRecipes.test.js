import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import ViewRecipes from '../../components/recipes/ViewRecipes';


describe('ViewRecipes component', () => {
  const pathname = 'path';
  const props = {
    location: { pathname },

    match: {
      params: {
        category_id: 1
      }
    },
  };
  const del = jest.fn();
  const wrapper = shallow(<ViewRecipes  {...props}/>);

  it('renders properly without crashing', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('has clickable buttons', () => {
    expect(wrapper.find('#editRec').length).toBe(0);
  });


});
