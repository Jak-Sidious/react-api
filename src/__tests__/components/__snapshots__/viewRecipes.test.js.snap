// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ViewRecipes /> should render properly 1`] = `
<div
  className="mainBackground"
>
  <Navigation
    pathname="path"
  />
  <_class
    options={Object {}}
  />
  <ModalEditCat
    closeModal={[Function]}
    desc=""
    handleChange={[Function]}
    handleEdit={[Function]}
    name=""
    showModal={false}
  />
  <ModalCreateRecipe
    category_name=""
    closeModal1={[Function]}
    handleChange={[Function]}
    handleCreate={[Function]}
    redirectRecipes={[Function]}
    showModal1={false}
  />
  <br />
  <div>
    <Grid
      columns={3}
      container={true}
    >
      <form
        onSubmit={[Function]}
      >
        <input
          className="catSearch"
          name="search"
          onChange={[Function]}
          placeholder="Search.."
          type="search"
        />
      </form>
      <GridRow />
      <Button
        animated={true}
        as="button"
        className="left floated"
        onClick={[Function]}
      >
        <ButtonContent
          visible={true}
        >
          Previous
        </ButtonContent>
        <ButtonContent
          hidden={true}
        >
          <Icon
            as="i"
            name="left arrow"
          />
        </ButtonContent>
      </Button>
      <Button
        animated={true}
        as="button"
        className="right floated"
        onClick={[Function]}
      >
        <ButtonContent
          visible={true}
        >
          Next
        </ButtonContent>
        <ButtonContent
          hidden={true}
        >
          <Icon
            as="i"
            name="right arrow"
          />
        </ButtonContent>
      </Button>
    </Grid>
    <h1>
       
      You currently do not have any categories please create a few
       
    </h1>
  </div>
</div>
`;
